### PR TITLE
[ACM-10578] WIP: partial match

### PR DESCRIPTION
**Pull request by SherinV**
### Related Issue

https://issues.redhat.com/browse/ACM-10578

### Description of changes
- Added ...

### CODE REVIEW

**Comment by SherinV** [comment]
Pending: Have to add test cases.

File: pkg/resolver/search.go [hunk 1]
```diff
@@ -316,45 +316,51 @@ func WhereClauseFilter(ctx context.Context, input *model.SearchInput,
...
+				opValueMap = getOperatorIfDateFilter(filter.Property, values, opValueMap)
+				simpleMatch = false
+			}
+			if compareValues(values, []string{"*"}) {
+				opValueMap = getPartialMatchFilter(filter.Property, values, dataType, opValueMap)
+				simpleMatch = false
+			}
+			if simpleMatch {
+				opValueMap = extractOperator(values, "", opValueMap)
+			}
```
Comment (jlpadilla) [inline pkg/resolver/search.go:354]: 1. Can we use `if/else if` to make it clear that only 1 should match?
2. Maybe this block could be moved to a function.

**Comment by jlpadilla** [comment]
Let's use this opportunity to document the supported operations and the data types it applies to.
Here's a start:

Operation      | Data type                | Comments
---                  | ---                           | ---
=                     | string, number       | This is the default operation.
! or !=             | string, number       | NOT operation.
<, <=, >, >=   | number                   | 
TODO             | date                        | Dates matching the last hour, day, week, month, year
 \*                     | string                     |  Partial string match.

File: pkg/resolver/search.go (hunk 1 above)
Comment (SherinV) [reply pkg/resolver/search.go]: Done

File: pkg/resolver/searchHelper.go [hunk 2]
```diff
@@ -24,33 +23,61 @@ func getPropertyType(ctx context.Context, refresh bool) (map[string]string, erro
...
-		">=", "",
-		"!=", "",
-		"!", "",
-		"<", "",
-		">", "",
-		"=", "")
-	operatorValue := map[string][]string{}
+// Extract operator (<=, >=, !=, !, <, >, =) if any from string
+func getOperatorFromString(value string) (string, string) {
+	klog.Info("value: ", value)
```
Comment (jlpadilla) [inline pkg/resolver/searchHelper.go:28]: please lower verbosity

File: pkg/resolver/searchHelper.go [hunk 3]
```diff
@@ -24,33 +23,61 @@ func getPropertyType(ctx context.Context, refresh bool) (map[string]string, erro
...
+	operator := "="
+	operand := value
+
+	prefixes := []string{"<=", ">=", "!=", "!", "<", ">", "="}
+	for _, prefix := range prefixes {
+		if cutString, yes := strings.CutPrefix(value, prefix); yes {
+			operator = prefix
+			operand = cutString
+			klog.Info("operator: ", operator)
+			klog.Info("operand: ", operand)
```
Comment (jlpadilla) [inline pkg/resolver/searchHelper.go:38]: Lower verbosity.  Consider combining all 3 logs in this function into a single log.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**